coffee-shop-application/src/main/kotlin/app/routes/OrderRoutes.kt

```diff
@@ -0,0 +1,239 @@
+package app.routes
+import app.models.*
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import io.ktor.http.*
+import io.ktor.server.request.receive
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.transactions.transaction
+import kotlinx.serialization.Serializable
+import app.dto.OrderCreateRequest
+import app.dto.OrderItemRequest
+import app.dto.OrderUpdateRequest
+import io.ktor.server.auth.jwt.JWTPrincipal
+import io.ktor.server.auth.principal
+
+
+fun Route.orderRoutes() {
+    route("/orders") {
+
+        get {
+            val principal = call.principal<JWTPrincipal>()
+            val role = principal?.payload?.getClaim("role")?.asString()
+
+            if (role != "CUSTOMER" && role != "STAFF") {
+                call.respond(HttpStatusCode.Forbidden, "Unauthorized")
+                return@get
+            }
```

Comment:
You authenticated using JWT, that's correct, but I would move authorization to business role specific actions to service layer. The router should not have any business logic, and roled base access is logic.